mage_ai/cache/tag.py

```diff
@@ -144,6 +144,9 @@ def remove_pipeline(self, tag_uuid: str, pipeline_uuid: str, repo_path: str = No
 
         mapping[key][KEY_FOR_PIPELINES] = pipelines_dict
 
+        if not pipelines_dict and mapping[key].keys() == set([KEY_FOR_PIPELINES]):
```

Comment:
This checks if no pipelines are still using the tag, and if there aren't any, it'll purge it from the cache (so it doesn't still appear in the pipeline filters dropdown).

Reply:
this comparison does look weird to me, but i can't think of a better way to do it right now